client/components/bulk-checkbox/index.js

```diff
@@ -11,9 +12,9 @@ const BulkCheckbox = ( { selectedCount, allCount, onToggle } ) => {
 	};
 
 	return (
-		<span className="wcc-bulk-checkbox-container" onClick={ onClick }>
-			<input type="checkbox" checked={ allSelected } readOnly />
-			{ someSelected ? <Gridicon className="bulk-select__some-checked-icon" icon="minus-small" size={ 18 }/> : null }
+		<span className="form-checkbox bulk-checkbox" onClick={ onClick }>
+			<Checkbox checked={ allSelected } readOnly />
+			{ someSelected && <Gridicon icon="minus-small" size={ 16 } /> }
```

Comment:
You're totally right. I've consolidated all the "indeterminate" functionality into the Checkbox component. This is ready for review now :)